networkx/algorithms/connectivity/kcutsets.py

```diff
@@ -0,0 +1,221 @@
+            flow_value = R.graph['flow_value']
+
+            if flow_value == k:
+                ## Remove saturated edges form the residual network
+                saturated_edges = list((u, w, d) for (u, w, d) in 
+                                        R.edges_iter(data=True)
+                                        if d['capacity'] == d['flow'])
+                R.remove_edges_from(saturated_edges)
+                # step 6: shrink the strongly connected components of 
+                # residual flow network R and call it L
+                L = nx.condensation(R)
+                cmap = L.graph['mapping']
+                # step 7: Compute antichains of L; they map to closed sets in H
+                # Any edge in H that links a closed set is part of a cutset
+                for antichain in antichain_generator(L):
+                    # Nodes in an antichain of the condensation graph of
+                    # the residual network map to a closed set of nodes that
+                    # define a node partition of the auxiliary digraph H.
+                    S = set(n for n, scc in cmap.items() if scc in antichain)
+                    # Find the cutset that links the node partition (S,~S) in H
+                    cutset = set()
+                    for u, nbrs in ((n, H[n]) for n in S):
+                        cutset.update((u, w) for w in nbrs if w not in S)
+                    # The edges in H that form the cutset are internal edges
+                    # (ie edges that represent a node of the original graph G)
+                    node_cut = set(H.node[n]['id'] for edge in cutset for n in edge)
+
+                    if len(node_cut) == k:
+                        if node_cut not in seen:
+                            yield node_cut
+                            seen.append(node_cut)
+                        # Add an edge (x, v) to make sure that we do not
+                        # find this cutset again. This is equivalent
+                        # of adding the edge in the input graph 
+                        # G.add_edge(x, v) and then regenerate H and R:
+                        # Add edges to the auxiliary digraph.
+                        H.add_edge('%sB' % mapping[x], '%sA' % mapping[v],
+                                   capacity=1)
+                        H.add_edge('%sB' % mapping[v], '%sA' % mapping[x],
+                                   capacity=1)
+                        # Add edges to the residual network.
+                        R.add_edge('%sB' % mapping[x], '%sA' % mapping[v],
+                                   capacity=1)
+                        R.add_edge('%sA' % mapping[v], '%sB' % mapping[x],
+                                   capacity=1)
+                        break
+                # Add again the saturated edges to reuse the residual network
+                R.add_edges_from(saturated_edges)
+
+
+def transitive_closure(G):
```

Comment:
It is surprising that we did not have this function. We should put this in an appropriate location and formally expose it.